security/cmd/istio_ca/main.go

```diff
@@ -252,11 +379,38 @@ func createCA(core corev1.SecretsGetter) ca.CertificateAuthority {
 	}
 
 	caOpts.LivenessProbeOptions = opts.LivenessProbeOptions
+	caOpts.ProbeCheckInterval = opts.probeCheckInterval
 
 	istioCA, err := ca.NewIstioCA(caOpts)
 	if err != nil {
 		log.Errorf("Failed to create an Istio CA (error: %v)", err)
 	}
+
+	if opts.LivenessProbeOptions.IsValid() {
+		livenessProbe := probe.NewProbe()
+		livenessProbeController := probe.NewFileController(opts.LivenessProbeOptions)
+		livenessProbe.RegisterProbe(livenessProbeController, "liveness")
+		livenessProbeController.Start()
+
+		// set initial status to good
+		livenessProbe.SetAvailable(nil)
+
+		livenessProbeChecker := &LivenessCheckController{
+			rootCertFile: opts.rootCertFile,
+			interval:     opts.probeCheckInterval,
+			grpcHostname: opts.grpcHostname,
+			grpcPort:     opts.grpcPort,
+			rsaKeySize:   2048,
+
+			livenessProbe: livenessProbe,
+
+			ca:        istioCA,
+			caCertTTL: opts.caCertTTL,
+			client:    &grpcclient.CAGrpcClientImpl{},
+		}
+		livenessProbeChecker.Run()
```

Comment:
Not running this as a go routine?

Reply:
There is a go routine inside the Run() method